### PR TITLE
Resolve #3, #6 and #14

### LINKED ISSUE #3: Add 'parser' argument for generating standalone HTML code
The generated HTML relies on the content of the `css/` folder (css, fonts and images).
Add an argument to allow the generation of HTML code which points to remote css, fonts and images.

P.S. #6 (Create a 'theme' file and add '--theme' argument) should help to solve this.

### PR DESCRIPTION
**Pull request by arthurazs**
Create 2 `templates`:
- **default** (points to `gh-pages` css folder)
- **dota2** (points to `dota2.com` css files)

Also add `--template` argument. If omitted, uses **default** template.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/arthurazs/dota2patches/pull/16" target="_blank">ReviewNinja

File: model.py [hunk 1]
```diff
@@ -1,46 +1,22 @@
 from __future__ import print_function
 from data import HeropediaData
+import ast
 
 class Html (object):
 
+    _TEMPLATES_FOLDER = 'templates/'
+
+    def _read_template(self, template_name):
+        with open(self._TEMPLATES_FOLDER + template_name, 'r') as template_file:
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (80 > 79 characters)

File: model.py [hunk 2]
```diff
@@ -1,46 +1,22 @@
...
+            template_dictionary = ast.literal_eval(template_file.read())
+        return template_dictionary
+
     #Initialization
-    def __init__(self, title):
+    def __init__(self, title, template = 'default'):
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 3]
```diff
@@ -1,46 +1,22 @@
...
-
-        <a class="HiddenAnchor" name="GameplayHeader"> </a>
-        <div class="Container" id="Header">
-            <div class="Inner" style="text-align:center">
-                <h1>Gameplay Update {0}</h1>
-                <h3><a href="index.html">Previous changelogs</a></h3>
-            </div>
-        </div>
-'''.format(self._title)
+        self._content = self._template_dictionary['OPEN_HTML'].format(title = self._title)
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E251] unexpected spaces around keyword / parameter equals
- [E501] line too long (90 > 79 characters)

File: model.py [hunk 4]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
     #Add GENERAL Contents
     def addGeneral(self, lines):
         if(lines):
-            self._addContent(r'''
-        <div class="Container RepeatY BGStyle{}">
-            <div class="Inner">
-                <h3>General</h3>'''.format(int(self._bgStyle)))
+            self._addContent(self._template_dictionary['OPEN_GENERAL'].format(style = int(self._bgStyle)))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (106 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 5]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
...
-                self._addContent(r'''
-                    <li>{}</li>'''.format(line))
-            self._addContent(r'''
-                    </ul>
-''')
-
-            self._addContent(r'''
-            </div>
-        </div>''')
+                self._addContent(self._template_dictionary['GENERAL_LI'].format(line = line))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (93 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 6]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
...
             self._bgStyle = not self._bgStyle
 
     #Add ITEMS Contents
     def addItems(self, dictionary):
         if(dictionary):
-            self._addContent(r'''
-        <div class="Container RepeatY BGStyle{}" id="Items">
-            <div class="Inner">
-                <h3>Item changes</h3>'''.format(int(self._bgStyle)))
+            self._addContent(self._template_dictionary['OPEN_ITEMS'].format(style = int(self._bgStyle)))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (104 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 7]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
...
 
             for key, values in sorted(dictionary.items()):
-                self._addContent(r'''
-
-                    [[{}]]
-                    <ul>'''.format(key))
+                self._addContent(self._template_dictionary['OPEN_ITEMS_UL'].format(key = key))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (94 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 8]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
...
-                    self._addContent(r'''
-                        <li>{}</li>'''.format(value))
-                self._addContent(r'''
-                    </ul>
-''')
-
-            self._addContent(r'''
-            </div>
-        </div>''')
+                    self._addContent(self._template_dictionary['ITEMS_LI'].format(value = value))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (97 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 9]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
...
             self._bgStyle = not self._bgStyle
 
     #Add HERO Contents
     def addHeros(self, dictionary):
         if(dictionary):
-            self._addContent(r'''
-        <div class="Container RepeatY BGStyle{}" id="Heroes">
-             <div class="Inner">
-                <h3>Heroes</h3>'''.format(int(self._bgStyle)))
+            self._addContent(self._template_dictionary['OPEN_HEROES'].format(style = int(self._bgStyle)))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (105 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 10]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
...
 
             for key, values in sorted(dictionary.items(), key=HeropediaData.sort):
-                self._addContent(r'''
-
-                    [[{}]]
-                    <ul>'''.format(key))
+                self._addContent(self._template_dictionary['OPEN_HEROES_UL'].format(key = key))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (95 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: model.py [hunk 11]
```diff
@@ -49,84 +25,47 @@ def _addContent(self, text):
...
-                    self._addContent(r'''
-                        <li>{}</li>'''.format(value))
-                self._addContent(r'''
-                    </ul>
-''')
-
-            self._addContent(r'''
-            </div>
-        </div>''')
+                    self._addContent(self._template_dictionary['HEROES_LI'].format(value = value))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E501] line too long (98 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: patch.py [hunk 12]
```diff
@@ -8,12 +8,22 @@
 
 parser = argparse.ArgumentParser(description="This software formats a Dota2' changelog text into HTML.")
 parser.add_argument('--file', '-f', action='store', help="changelog to be formated", required = True, dest = 'file')
+parser.add_argument('--template', '-t', action='store', help="base template to generate HTML", default = 'default', dest = 'template')
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E501] line too long (134 > 79 characters)
- [E251] unexpected spaces around keyword / parameter equals

File: patch.py [hunk 13]
```diff
@@ -8,12 +8,22 @@
...
 parser.add_argument('--version', '-v', action='version', version='%(prog)s: v1.0 (Yasha)')
 args = parser.parse_args()
 
 #CONSTANT
 CHANGELOG = 'changelogs/'
 
+#Checks if the line starts with fixed and removes it
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E265] block comment should start with '# '

File: patch.py [hunk 14]
```diff
@@ -8,12 +8,22 @@
...
+def getName(value):
```
Comment (arthurazs) [inline patch.py:31]: **[flake8]**
- [E302] expected 2 blank lines, found 1

File: model.py [hunk 15]
```diff
@@ -1,46 +1,24 @@
...
 from data import HeropediaData
+import ast
 
 class Html (object):
 
+    _TEMPLATES_FOLDER = 'templates/'
+
+    def _read_template(self, template_name):
+        with open(self._TEMPLATES_FOLDER + template_name, 'r') \
+        as template_file:
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 16]
```diff
@@ -1,46 +1,24 @@
...
-        <a class="HiddenAnchor" name="GameplayHeader"> </a>
-        <div class="Container" id="Header">
-            <div class="Inner" style="text-align:center">
-                <h1>Gameplay Update {0}</h1>
-                <h3><a href="index.html">Previous changelogs</a></h3>
-            </div>
-        </div>
-'''.format(self._title)
+        self._content = self._template_dictionary['OPEN_HTML'] \
+        .format(title=self._title)
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 17]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
     #Add GENERAL Contents
     def addGeneral(self, lines):
         if(lines):
-            self._addContent(r'''
-        <div class="Container RepeatY BGStyle{}">
-            <div class="Inner">
-                <h3>General</h3>'''.format(int(self._bgStyle)))
-
-            self._addContent(r'''
+            self._addContent(self._template_dictionary['OPEN_GENERAL'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E502] the backslash is redundant between brackets

File: model.py [hunk 18]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+            .format(style=int(self._bgStyle)))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: model.py [hunk 19]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+            self._addContent(
+            self._template_dictionary['OPEN_GENERAL_UL'])
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 20]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
-                    <li>{}</li>'''.format(line))
-            self._addContent(r'''
-                    </ul>
-''')
-
-            self._addContent(r'''
-            </div>
-        </div>''')
+                self._addContent(
+                self._template_dictionary['GENERAL_LI'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented
- [E502] the backslash is redundant between brackets

File: model.py [hunk 21]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                .format(line=line))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 22]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+
+            self._addContent(
+            self._template_dictionary['CLOSE_GENERAL_UL'])
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 23]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
             self._bgStyle = not self._bgStyle
 
     #Add ITEMS Contents
     def addItems(self, dictionary):
         if(dictionary):
-            self._addContent(r'''
-        <div class="Container RepeatY BGStyle{}" id="Items">
-            <div class="Inner">
-                <h3>Item changes</h3>'''.format(int(self._bgStyle)))
+            self._addContent(self._template_dictionary['OPEN_ITEMS'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E502] the backslash is redundant between brackets

File: model.py [hunk 24]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+            .format(style=int(self._bgStyle)))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: model.py [hunk 25]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
 
             for key, values in sorted(dictionary.items()):
-                self._addContent(r'''
-
-                    [[{}]]
-                    <ul>'''.format(key))
+                self._addContent(
+                self._template_dictionary['OPEN_ITEMS_UL'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented
- [E502] the backslash is redundant between brackets

File: model.py [hunk 26]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                .format(key=key))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 27]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
-                        <li>{}</li>'''.format(value))
-                self._addContent(r'''
-                    </ul>
-''')
-
-            self._addContent(r'''
-            </div>
-        </div>''')
+                    self._addContent(
+                    self._template_dictionary['ITEMS_LI'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented
- [E502] the backslash is redundant between brackets

File: model.py [hunk 28]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                    .format(value=value))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 29]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                self._addContent(
+                self._template_dictionary['CLOSE_ITEMS_UL'])
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 30]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
-        <div class="Container RepeatY BGStyle{}" id="Heroes">
-             <div class="Inner">
-                <h3>Heroes</h3>'''.format(int(self._bgStyle)))
-
-            for key, values in sorted(dictionary.items(), key=HeropediaData.sort):
-                self._addContent(r'''
-
-                    [[{}]]
-                    <ul>'''.format(key))
+            self._addContent(self._template_dictionary['OPEN_HEROES'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E502] the backslash is redundant between brackets

File: model.py [hunk 31]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+            .format(style=int(self._bgStyle)))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: model.py [hunk 32]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+
+            for key, values in sorted(dictionary.items(),
+            key=HeropediaData.sort):
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: model.py [hunk 33]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                self._addContent(
+                self._template_dictionary['OPEN_HEROES_UL'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented
- [E502] the backslash is redundant between brackets

File: model.py [hunk 34]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                .format(key=key))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 35]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
-                        <li>{}</li>'''.format(value))
-                self._addContent(r'''
-                    </ul>
-''')
-
-            self._addContent(r'''
-            </div>
-        </div>''')
+                    self._addContent(
+                    self._template_dictionary['HEROES_LI'] \
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented
- [E502] the backslash is redundant between brackets

File: model.py [hunk 36]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                    .format(value=value))
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: model.py [hunk 37]
```diff
@@ -49,84 +27,65 @@ def _addContent(self, text):
...
+                self._addContent(
+                self._template_dictionary['CLOSE_HEROES_UL'])
```
Comment (arthurazs) [inline model.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: patch.py [hunk 38]
```diff
@@ -6,23 +6,40 @@
 from model import Html
 from data import HeropediaData
 
-parser = argparse.ArgumentParser(description="This software formats a Dota2' changelog text into HTML.")
-parser.add_argument('--file', '-f', action='store', help="changelog to be formated", required = True, dest = 'file')
-parser.add_argument('--version', '-v', action='version', version='%(prog)s: v1.0 (Yasha)')
+
+parser = argparse.ArgumentParser(
+description="This software formats a Dota2' changelog text into HTML.")
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: patch.py [hunk 39]
```diff
@@ -6,23 +6,40 @@
...
+parser.add_argument('--file', '-f', action='store',
+help="changelog to be formated", required=True, dest='file')
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: patch.py [hunk 40]
```diff
@@ -6,23 +6,40 @@
...
+parser.add_argument('--template', '-t', action='store',
+help="base template to generate HTML", default='default',
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: patch.py [hunk 41]
```diff
@@ -6,23 +6,40 @@
...
+dest='template')
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: patch.py [hunk 42]
```diff
@@ -6,23 +6,40 @@
...
+parser.add_argument('--version', '-v', action='version',
+version='%(prog)s: v1.0 (Yasha)')
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: patch.py [hunk 43]
```diff
@@ -65,24 +74,25 @@ def getName(value):
...
         model.addItems(item)
         model.addHeros(hero)
         model.close()
         print(model.getContent(), file=text)
 
-    #Feedback
-    currentLineCount = sum(len(changes) for changes in hero.values()) + sum(len(changes) for changes in item.values())
+    # Feedback
+    currentLineCount = sum(len(changes) for changes in hero.values()) \
+    + sum(len(changes) for changes in item.values())
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E122] continuation line missing indentation or outdented

File: patch.py [hunk 44]
```diff
@@ -93,15 +103,18 @@ def getName(value):
         if (status == 1):
             print('1 line under GENERAL updates:')
             print('* ' + ' '.join(lines))
-            print('\nThis line might be a hero/item update and you should manually place it at the proper location.')
+            print('\nThis line might be a hero/item update and you ' + \
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E502] the backslash is redundant between brackets

File: patch.py [hunk 45]
```diff
@@ -93,15 +103,18 @@ def getName(value):
...
+            'should manually place it at the proper location.')
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

File: patch.py [hunk 46]
```diff
@@ -93,15 +103,18 @@ def getName(value):
...
         else:
             print(str(status) + ' lines under GENERAL updates:')
             for line in lines:
                 print('* ' + line)
-            print('\nSome of these lines might be hero/item updates and you should manually place them at the proper location.')
+            print('\nSome of these lines might be hero/item updates' + \
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E502] the backslash is redundant between brackets

File: patch.py [hunk 47]
```diff
@@ -93,15 +103,18 @@ def getName(value):
...
+            ' and you should manually place them at the proper ' + \
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent
- [E502] the backslash is redundant between brackets

File: patch.py [hunk 48]
```diff
@@ -93,15 +103,18 @@ def getName(value):
...
+            'location.')
```
Comment (arthurazs) [inline patch.py]: **[flake8]**
- [E128] continuation line under-indented for visual indent

**Pull request merged**